File: board-room/src/main/java/net/stzups/board/server/websocket/protocol/client/messages/ClientMessageOpenDocument.java
package net.stzups.board.server.websocket.protocol.client.messages;

import io.netty.buffer.ByteBuf;
import net.stzups.board.server.websocket.protocol.client.ClientMessage;
import net.stzups.board.server.websocket.protocol.client.ClientMessageType;

public class ClientMessageOpenDocument extends ClientMessage {
    private long id;

    public ClientMessageOpenDocument(ByteBuf byteBuf) {
        super(ClientMessageType.OPEN_DOCUMENT);
        this.id = byteBuf.readLong();
    }

    public long getId() {
        return id;
    }
}

File: board-frontend/html/scripts/protocol/client/messages/ClientMessageOpenDocument.js
import ClientMessageType from "../ClientMessageType.js";
import ClientMessage from "../ClientMessage.js";

export default class ClientMessageOpenDocument extends ClientMessage {
    constructor(document) {
        super(ClientMessageType.OPEN_DOCUMENT);
        this.id = document.id;
    }

    getBufferSize() {
        return super.getBufferSize() + 8;
    }

    serialize(writer) {
        super.serialize(writer);
        writer.writeBigInt64(this.id);
    }
}